Review advanced-use.rst

Mrt134 opened the pull request:
(Ignored text accuracy for this commit)
Updated advanced-use.rst with:
- Grammar fixes
- Syntax fixes (ReST/Sphinx)
- Emphasis on important words.
- Rewrite/reorganize sentences for clarity.
- PEP8 fixes.
- White spacing fixes.

Make sure to follow and check these boxes before submitting a PR! Thank you.

- [x] Explain the feature
- [x] Read [CONTRIBUTING.md](https://github.com/iocage/iocage/blob/master/CONTRIBUTING.md)

Mrt134 committed:
Review advanced-use.rst

(Ignored text accuracy for this commit)
Updated advanced-use.rst with:
- Grammar fixes
- Syntax fixes (ReST/Sphinx)
- Emphasis on important words.
- Rewrite/reorganize sentences for clarity.
- PEP8 fixes.
- White spacing fixes.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+To clone **www01** to **www02**, run:
 
-``iocage clone www01 tag=www02``
+:samp:`# iocage clone www01 tag=www02`
```
TAGS are being dropped, use `--name` instead

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+Clone a jail from an existing snapshot with:
 
-``iocage clone www01@snapshotname tag=www03``
+:samp:`# iocage clone www01@snapshotname tag=www03`
```
Same as above

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+**To promote a cloned jail, run:**
 
-``iocage promote UUID | TAG``
+:samp:`# iocage promote UUID | TAG`
```
s/TAG/NAME/, but make it clear promote isn't in yet.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+**The demoted jail can now be removed:**
 
-``iocage destroy UUID | TAG``
+:samp:`# iocage destroy UUID | TAG`
```
s/TAG/NAME/

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+To update a jail to latest patch level, run:
 
-``iocage update UUID | TAG``
+:samp:`# iocage update UUID | TAG`
```
s/TAG/NAME/

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+remove the snapshot:
 
-``iocage snapremove UUID|TAG@snapshotname``
+:samp:`# iocage snapremove UUID|TAG@snapshotname`
```
s/TAG/NAME/ and the user is now responsible for the snapshots, that may change in the future if I port that functionality back, but currently we don't do it automatically.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+If the update breaks the jail, revert back to the original snapshot:
 
-``iocage rollback UUID|TAG@snapshotname``
+:samp:`# iocage rollback UUID|TAG@snapshotname`
```
Drop this and the previous line.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+To clone a jail, run:
 
-``iocage clone UUID|TAG tag=testupdate``
+:samp:`# iocage clone UUID|TAG tag=testupdate`
```
s/TAG/NAME/ and use `--name` not `tag`.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
-to the hosts RELEASE version (uname -r).
+Upgrades are handled with the freebsd-update(8) utility. By default, the
+upgrade command attempts to upgrade the jail to the host's RELEASE
+version (visible with uname -r).
```
No longer true, the user must supply `--release`

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+To upgrade a normal jail (non-basejail) to the host's RELEASE, run:
 
-``iocage upgrade UUID | TAG``
+:samp:`# iocage upgrade UUID | TAG`
```
`iocage upgrade -r 11.1-RELEASE UUID | NAME`

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+To upgrade to a specific release, run:
 
-``iocage upgrade UUID|TAG release=10.1-RELEASE``
+:samp:`# iocage upgrade UUID|TAG release=10.1-RELEASE`
```
Drop this and previous 2 lines

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+:samp:`# iocage upgrade UUID|TAG release=10.1-RELEASE`
 
 Upgrade basejail
 ++++++++++++++++
```
The entire basejail section needs a disclaimer that this upgrade mechanism for them is not written yet. So possibly remove it entirely or make it known that this is only for the old legacy implementation for now.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+To enable a jail to auto-boot during a system boot, simply run:
 
-``iocage set boot=on UUID|TAG``
+:samp:`# iocage set boot=on UUID|TAG`
```
s/TAG/NAME/

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
 Boot order can be specified by setting the priority value:
 
-``iocage set priority=20 UUID|TAG``
+:samp:`# iocage set priority=20 UUID|TAG`
```
s/TAG/NAME/

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
 List snapshots for a jail with:
 
-``iocage snaplist UUID|TAG``
+:samp:`# iocage snaplist UUID|TAG`
```
s/TAG/NAME/

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+To create a new snapshot, run:
 
-``iocage snapshot UUID|TAG``
+:samp:`# iocage snapshot UUID|TAG`
```
`iocage snapshot -n NAME UUID|NAME` we require they supply a name now.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+To create a snapshot with custom naming, run:
 
-``iocage snapshot UUID|TAG@mysnapshotname``
+:samp:`# iocage snapshot UUID|TAG@mysnapshotname`
```
Remove this and previous 2 lines.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+:samp:`# iocage snapshot UUID|TAG@mysnapshotname`
 
 Resource limits
 ---------------
```
Disclaimer that this is only for legacy implementations, it has not been written for the new version as of yet.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
-containing one package name per line. Please note you need to have Internet
-connection for this to work as ``pkg install`` will try to get the packages from
+Specify the ``pkglist`` property at creation time, which needs to point
+to a text file containing one package name per line. Note an Internet
```
to a JSON file.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
 **Example:**
 
-Create a pkgs.txt file and add package names to it.
+Create a :file:`pkgs.txt` file and add package names to it.
```
This section needs the pkglist I gave you before, it's not a plain text file.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+   tmux
 
-Now simply create a jail and supply the pkgs.txt file:
+Now, create a jail and supply :file:`pkgs.txt`:
```
`.json`

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -6,246 +6,256 @@ Clones
+Now, create a jail and supply :file:`pkgs.txt`:
 
-``iocage create pkglist=/path-to/pkgs.txt tag=myjail``
+:samp:`# iocage create pkglist=/path-to/pkgs.txt tag=myjail`
```
`iocage create `--pkglist /path/to/pkg.json --name NAME`

Mrt134 committed:
Update advanced-use.rst

- Address all review comments.
- Add index and internal reference hooks.
- Adjust the :command: and :samp: directives to show examples vs straight commands.
- Add brackets to commands to show replaceable elements.
- Fix a few spelling/format errors.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -11,251 +16,317 @@ Essentially, clones are cheap, lightweight, and writable snapshots.
 To clone a jail, run:
 
-:samp:`# iocage clone UUID|TAG tag=testupdate`
+:command:`iocage clone [UUID|NAME] --name [testupdate]`
```
We should note that `--name` is optional in this case. If they don't supply it, they will just get a UUID named jail.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -11,251 +16,317 @@ Essentially, clones are cheap, lightweight, and writable snapshots.
+List snapshots for a jail:
 
-:samp:`# iocage snaplist UUID|TAG`
+:command:`iocage snaplist [UUID|TAG]`
```
s/TAG/NAME/

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -11,251 +16,317 @@ Essentially, clones are cheap, lightweight, and writable snapshots.
+Create a new snapshot:
 
-:samp:`# iocage snapshot UUID|TAG`
+:command:`iocage snapshot [UUID|TAG]`
```
s/TAG/NAME/

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -11,251 +16,317 @@ Essentially, clones are cheap, lightweight, and writable snapshots.
+.. _Reset Jail Properties:
 
-iocage easily allows resetting a jail's properties back to their
+Reset Jail Properties
```
`reset` does not exist anymore. This whole entry can be dropped.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -11,251 +16,317 @@ Essentially, clones are cheap, lightweight, and writable snapshots.
+Now, create a jail and supply :file:`pkgs.json`:
 
-:samp:`# iocage create pkglist=/path-to/pkgs.txt tag=myjail`
+:command:`iocage create -r [RELEASE] -p [path-to/pkgs.json] -n [NAME]`
```
Should note that `--name` is not required, a UUID jail will be created otherwise.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -11,251 +16,317 @@ Essentially, clones are cheap, lightweight, and writable snapshots.
+:file:`pkgs.json`:
 
 .. code-block:: none
 
```
I'm not sure if sphinx will automatically put the `{}` and such to make this valid JSON, but if it doesn't we need to make sure the example is proper JSON.

EX:
```
{
    "pkgs": [
    "nginx",
    "tmux"
    ]
}
```

Mrt134 committed:
Add requested review changes.

- Fix those last TAG errors.
- Add notes to show the -n flag is optional.
- Remove "reset jails" section.
- change pkgs.json code-block to render json and add syntax.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -81,6 +81,9 @@ To clone a jail, run:
 :command:`iocage clone [UUID|NAME] --name [testupdate]`
 
+.. note:: The **--name** flag is optional. :command:`iocage` assigns a
+   new UUID to the clone regardless of **--name** usage.
```
No, we don't assign a UUID if they use `--name`, if they don't we do.

Example:
`iocage create -n foo -r 11.0-RELEASE` will create a jail named `foo`.
`iocage create -r 11.0-RELEASE` will create a jail named `fd401c05-7777-44e7-9d83-289839f26351` for example.

skarekrow commented on doc/source/advanced-use.rst:
```diff
@@ -320,13 +301,20 @@ Create a :file:`pkgs.json` file and add package names to it.
 :command:`iocage create -r [RELEASE] -p [path-to/pkgs.json] -n [NAME]`
 
+.. note:: The **-n** flag is optional. A UUID is assigned to the new
+   jail regardless of its use.
```
Same note as above :)

skarekrow commented:
Close! SO close timmm!

Mrt134 committed:
Fix up --name notes:

Reworked notes about optional --name usage to reflect the either/or nature of UUID creation.
Thanks for the update!

Mrt134 commented:
Hopefully, this time'll get it. I got confused with the previous functionality of tags. :)
